server/controllers/stock/reports/index.js

```diff
@@ -26,24 +30,106 @@ const stockValue = require('./stock/value');
 const stockAssignReceipt = require('./stock/assign_receipt');
 const stockRequisitionReceipt = require('../requisition/requisition.receipt');
 
+/**
+ * @function determineReceiptType
+ *
+ * @description
+ * Figures out the type of stock receipt from the document uuid.  This allows
+ * a uniform API for rendering all stock receipts.
+ */
+async function determineReceiptType(uuid, isDepotTransferExit = -1) {
+
+  // this is only used when you are rendering receipt for transfering between depots
+  // to determine which receipt to render - the exit or the entry.
+  const directionality = isDepotTransferExit > -1
+    ? `AND is_exit = ${isDepotTransferExit}` : '';
+
+  const sql = `
+    SELECT document_uuid, flux_id FROM stock_movement
+    WHERE document_uuid = ? ${directionality}
+    LIMIT 1;
+  `;
+
+  const row = await db.one(sql, db.bid(uuid));
+  return row.flux_id;
+}
+
+async function renderStockReceipt(req, res, next) {
+  try {
+    const documentUuid = req.params.uuid;
+
+    let isDepotTransferExit = -1;
+    if (req.query.is_depot_transfer_exit) {
+      isDepotTransferExit = Number(req.query.is_depot_transfer_exit);
+    }
+
+    const receiptType = await determineReceiptType(documentUuid, isDepotTransferExit);
+
+    let renderer;
+
+    switch (receiptType) {
+    case Stock.flux.FROM_PURCHASE:
+      renderer = stockEntryPurchaseReceipt;
+      break;
+
+    case Stock.flux.FROM_OTHER_DEPOT:
+      renderer = stockEntryDepotReceipt;
+      break;
+
+    case Stock.flux.FROM_ADJUSTMENT:
+      renderer = stockAdjustmentReceipt;
+      break;
+
+    case Stock.flux.FROM_INTEGRATION:
```

Comment:
There is no `break;` here

Reply:
I assumed that FROM_INTEGRATION and TO_INTEGRATION used the same receipt, so I combined them.  If they don't, what receipt does stock exit via integration use?